common/onepanel/base/postgres-database.yaml

```diff
@@ -0,0 +1,64 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: postgres
+  namespace: onepanel
+spec:
+  selector:
+    matchLabels:
+      app: postgres
+  serviceName: "postgres-svc"
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: postgres
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+        - name: postgres
+          image: postgres
+          ports:
+            - containerPort: 5432
+          env:
+            - name: POSTGRES_DB
+              value: "onepanel"
+            - name: POSTGRES_USER
+              valueFrom:
+                secretKeyRef:
+                  name: onepanel
+                  key: databaseUsername
+            - name: POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: onepanel
+                  key: databasePassword
+          volumeMounts:
+            - name: postgres-pv-claim
+              mountPath: /var/lib/postgresql/data
+              subPath: postgres
+  volumeClaimTemplates:
+    - metadata:
+        name: postgres-pv-claim
+        labels:
+          app: postgres
+      spec:
+        storageClassName: standard
```

Comment:
Where is this `StorageClass` defined? Is this what comes with minikube?

Reply:
![image](https://user-images.githubusercontent.com/10407733/75724319-001bef00-5c93-11ea-8062-f3656767ac48.png)
Correct. Think we should make this a param?? Or add our own?

Reply:
We need to add our own, because k3s and microk8s may not call theirs `standard`, we already have a storageclass, I think that's only for GKE right now.

Reply:
Sure, sounds good. Any thoughts what I should name it? `onepanel-default`?

Reply:
I think `onepanel` should suffice, it's already [here](https://github.com/onepanelio/manifests/blob/master/storage/base/storage-class.yaml) with this GCP overlay [here](https://github.com/onepanelio/manifests/blob/master/storage/overlays/gcp/storage-class.yaml). But it's called `default` which is confusing.

Reply:
I can rename those two if you'd like? To `onepanel`?

Reply:
cool, makes sense. You'd have to add a specific overlay for `minikube` and `microk8s`

Reply:
Ohhh..okay. Good point.
I'm not sure if our CLI checks those yet.

Reply:
Actually, just looked, for both microk8s and minikube we can use `default` storage class. For former you enable it via `microk8s.enable storage` and for latter it is already built in. cc @Vafilor 

For the cloud providers, we need to create SSD `StorageClass` and call it `onepanel` as they may already have `default` but that may be non-SSD storage and not sufficient for deep learning workflows.

Reply:
@rushtehrani @Vafilor Can you guys verify the microk8s storage class?
```NAME                          PROVISIONER            RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
microk8s-hostpath (default)   microk8s.io/hostpath   Delete          Immediate           false                  38m```
I thought it was this?

Reply:
That's what I got for microk8s. Let me verify default works.

Reply:
Default does not work out of the box.

```storageclass.storage.k8s.io "default" not found```

Reply:
should we explicitly ask for the `defaultStorage` class? I don't think we're going to distinguish between `microk8s` and `minikube`.  They both fall under `single-node`. Another option is to actually query the storageClass when `--provider` is `single-node` and use that value. For cloud providers, we create a SSD storageclass for them.

Reply:
Query would be good.

Why not distinguish between them? It can be a simple overlay in kustomize can't it?